Cake running on Mono can't load netstandard 2.0 assembly

Currently getting `You must add a reference to assembly 'netstandard, Version=2.0.0.0`

Minimal repro:
```
#addin nuget:?package=Microsoft.AspNetCore.TestHost&version=2.0.0&loaddependencies=true
#addin nuget:?package=Microsoft.AspNetCore.Http.Extensions&version=2.0.0&loaddependencies=true

using Microsoft.AspNetCore.Builder;
using Microsoft.Extensions.Configuration;

Information("Oh hi, Mark!");

new ConfigurationBuilder().Build();
```

Gives:
```
standard.cake(9,28): error CS0012: The type 'Object' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'.
```

--- a/src/Cake.Core/Scripting/ScriptConventions.cs
+++ b/src/Cake.Core/Scripting/ScriptConventions.cs
@@ -85,6 +85,19 @@ namespace Cake.Core.Scripting
             // This is just to please Roslyn when running under Mono. See issue https://github.com/dotnet/roslyn/issues/19364
             result.Add(_loader.Load(new AssemblyName("System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"))); // System.Runtime
             result.Add(_loader.Load(new AssemblyName("System.Collections, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"))); // System.Collections
+
+            try
+            {
+                result.Add(_loader.Load(new AssemblyName("netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51"))); // NETStandard.Library
+            }
+            catch
+            {
+                /*
+                 * Silently continue instead and let it blow up during runtime if netstandard assembly was not found.
+                 * TODO: Log that netstandard assembly was not found.
+                 * Unfortunately, logger is not available in this class, and that would be too big of a change for the 0.26.1 hotfix release.
+                 */
+            }
 #endif
 
             // Return the assemblies.

--- a/tests/integration/Cake.Core/Scripting/AddinDirective.cake
+++ b/tests/integration/Cake.Core/Scripting/AddinDirective.cake
@@ -1,0 +1,20 @@
+Task("Cake.Core.Scripting.AddinDirective.LoadNetStandardAddin")
+    .Does(() =>
+{
+    DotNetCorePack($"{Paths.Resources}/Cake.Core/Scripting/netstandard2.addin/netstandard2.addin.csproj",
+        new DotNetCorePackSettings {
+            Configuration = "Release"
+        });
+
+    var script = $@"#addin nuget:{Paths.Resources}/Cake.Core/Scripting/netstandard2.addin/bin/Release?package=netstandard2.addin&version=1.0.0
+        Information(""Magic number: {0}"", GetMagicNumber(false));
+        Information(""The answer to life: {0}"", TheAnswerToLife);
+    ";
+
+    CakeExecuteExpression(script);
+});
+
+//////////////////////////////////////////////////////////////////////////////
+
+Task("Cake.Core.Scripting.AddinDirective")
+    .IsDependentOn("Cake.Core.Scripting.AddinDirective.LoadNetStandardAddin");

--- a/tests/integration/build.cake
+++ b/tests/integration/build.cake
@@ -20,6 +20,7 @@
 #load "./Cake.Common/Tools/Cake/CakeAliases.cake"
 #load "./Cake.Common/Tools/DotNetCore/DotNetCoreAliases.cake"
 #load "./Cake.Common/Tools/NuGet/NuGetAliases.cake"
+#load "./Cake.Core/Scripting/AddinDirective.cake"
 #load "./Cake.Core/Scripting/DefineDirective.cake"
 #load "./Cake.Core/Scripting/LoadDirective.cake"
 #load "./Cake.Core/Scripting/SystemCollections.cake"
@@ -47,6 +48,7 @@ Setup(ctx =>
 //////////////////////////////////////////////////
 
 Task("Cake.Core")
+    .IsDependentOn("Cake.Core.Scripting.AddinDirective")
     .IsDependentOn("Cake.Core.Scripting.DefineDirective")
     .IsDependentOn("Cake.Core.Scripting.LoadDirective")
     .IsDependentOn("Cake.Core.Scripting.SystemCollections")

--- a/tests/integration/resources/Cake.Core/Scripting/netstandard2.addin/MyCakeExtension.cs
+++ b/tests/integration/resources/Cake.Core/Scripting/netstandard2.addin/MyCakeExtension.cs
@@ -1,0 +1,29 @@
+﻿using System;
+using Cake.Core;
+using Cake.Core.Annotations;
+
+public static class MyCakeExtension
+{
+    [CakeMethodAlias]
+    public static int GetMagicNumber(this ICakeContext context, bool value)
+    {
+        return value ? int.MinValue : int.MaxValue;
+    }
+
+    [CakeMethodAlias]
+    public static int GetMagicNumberOrDefault(this ICakeContext context, bool value, Func<int> defaultValueProvider = null)
+    {
+        if (value)
+        {
+            return int.MinValue;
+        }
+
+        return defaultValueProvider == null ? int.MaxValue : defaultValueProvider();
+    }
+
+    [CakePropertyAlias]
+    public static int TheAnswerToLife(this ICakeContext context)
+    {
+        return 42;
+    }
+}

--- a/tests/integration/resources/Cake.Core/Scripting/netstandard2.addin/netstandard2.addin.csproj
+++ b/tests/integration/resources/Cake.Core/Scripting/netstandard2.addin/netstandard2.addin.csproj
@@ -1,0 +1,11 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Cake.Core" Version="0.26.0" />
+  </ItemGroup>
+
+</Project>

